cocotb/share/lib/gpi_log/gpi_logging.c

```diff
@@ -99,6 +99,50 @@ const char *log_level(long level)
 #define LOG_SIZE    512
 static char log_buff[LOG_SIZE];
 
+
+/**
+ * Log without going through the python hook.
+ *
+ * This is needed for both when the hook isn't connected yet, and for when a
+ * python exception occurs while trying to use the hook.
+ */
+static void gpi_log_native_v(const char *name, long level, const char *pathname, const char *funcname, long lineno, const char *msg, va_list argp)
+{
+    if (level < GPIInfo) {
+        return;
+    }
+
+    int n = vsnprintf(log_buff, LOG_SIZE, msg, argp);
+    if (n < 0) {
+        fprintf(stderr, "Log message construction failed\n");
+    }
+
+    fprintf(stdout, "     -.--ns ");
+    fprintf(stdout, "%-9s", log_level(level));
+    fprintf(stdout, "%-35s", name);
+
+    n = strlen(pathname);
```

Comment:
Reusing variables is bad practice.

Reply:
Perhaps, but this is just old code that moved.